### ISSUE TITLE
Question about formatting input data

### ISSUE THREAD

**Issue by DanielFennhagen** [google/automl#1064]
Thanks for a great repo!

I'm trying to train efficientdet using the segmentation head.
My dataset was previously in coco format and using [create_coco_tfrecord.py](https://github.com/google/automl/blob/master/efficientdet/dataset/create_coco_tfrecord.py) it has been converted to tfrecords.

I followed the example in [segmentation.py](https://github.com/google/automl/blob/master/efficientdet/keras/segmentation.py) and my parse function returns an input image and it's coresponding masks with shapes [W, H, 3] and [W, H, NUM_CLASSES].

However, during training the loss keeps increasing with every epoch which leads me to believe either the input is wrongly formatted or I have set the model up incorrectly.

Following are a few snippets of what I'm doing:
<details closed>
<summary>Snippets</summary>
<br>

```
train_dataset = tf.data.TFRecordDataset(filename_tensor)
train_dataset = train_dataset.map(parse_example , num_parallel_calls=tf.data.experimental.AUTOTUNE)

train_dataset = train_dataset.cache().shuffle(1000).batch(batch_size).repeat()
train_dataset = train_dataset.prefetch(buffer_size=tf.data.experimental.AUTOTUNE)

config = hparams_config.get_efficientdet_config('efficientdet-d0') #num_classes and seg_num_classes = 17
config.heads = ['segmentation']
model = efficientdet_keras.EfficientDetNet(config=config)
model.build((1, 512, 512, 3))
model.compile(
    optimizer='adam',
    loss=tf.keras.losses.CategoricalCrossentropy(from_logits=True),
    metrics=['accuracy'])

model.fit(
    train_dataset,
    epochs=200,
    steps_per_epoch=steps_per_epoch,
    callbacks=[])

```

```
def parse_example(serialized_example):
  keys_to_features = {
        'image/encoded': tf.io.FixedLenFeature((), tf.string),
        ...
        'image/object/mask': tf.io.VarLenFeature(tf.string),
      }

  parsed_tensors = tf.io.parse_single_example(serialized=serialized_example, features=keys_to_features)

  for k in parsed_tensors:
    if isinstance(parsed_tensors[k], tf.SparseTensor):
      if parsed_tensors[k].dtype == tf.string:
        parsed_tensors[k] = tf.sparse.to_dense(parsed_tensors[k], default_value='')
      else:
        parsed_tensors[k] = tf.sparse.to_dense(parsed_tensors[k], default_value=0)

  image = decode_image(parsed_tensors)

  masks = decode_masks(parsed_tensors)
  masks = tf.transpose(masks, [1, 2, 0]) #end transpose
  return image, masks
```

```
def decode_masks(parsed_tensors):
  def _decode_png_mask(png_bytes):
    mask = tf.io.decode_png(png_bytes, channels=1, dtype=tf.uint8)
    mask = tf.image.resize(mask, [128, 128])
    mask = tf.squeeze(mask, axis=-1)
    mask.set_shape([None, None])
    mask = tf.math.ceil(mask) 
    return mask

  height = parsed_tensors['image/height']
  width = parsed_tensors['image/width']
  masks = parsed_tensors['image/object/mask']
  labels = parsed_tensors['image/object/class/label']

  indices = tf.reshape(labels, shape=[tf.shape(labels)[0], 1])
  updates = tf.cond(
    pred=tf.greater(tf.size(input=masks), 0),
    true_fn=lambda: tf.map_fn(_decode_png_mask, masks, dtype=tf.float32),
    false_fn=lambda: tf.zeros([0, height, width], dtype=tf.float32))

  shape = tf.constant([16, 128, 128], dtype=tf.int64)
  masks = tf.scatter_nd(indices, updates, shape)

  return masks
```
</details>

I am using CategoricalCrossentropy as my mask tensor is not sparse, is this a problem?
Do you guys have any idea of what I'm doing wrong?

Thanks

**Comment by keertika-11** [comment]
https://github.com/google/automl/issues/1013

**Comment by DanielFennhagen** [comment]
My data was in the wrong format, correct format can be read about here [keras losses](https://keras.io/api/losses/probabilistic_losses/#sparsecategoricalcrossentropy-class)
